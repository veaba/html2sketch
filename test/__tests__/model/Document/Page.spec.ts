import { Page } from 'html2sketch';
import { describe, test, expect, it, beforeAll  } from 'vitest'

describe('Page 类', () => {
  describe('Page 创建', () => {
    it('带参数创建', () => {
      const page = new Page({ width: 200, height: 100 });
      expect(page.toSketchJSON()).toMatchSnapshot();
    });
    it('不带参数创建', () => {
      const page = new Page();
      expect(page.toSketchJSON()).toMatchSnapshot();
    });
  });
  describe('toSketchJSON', () => {
    it('带参数生成', () => {
      const page = new Page({ width: 200, height: 100 });
      const json = page.toSketchJSON();
      expect(json._class).toBe('page');
      expect(json.frame.width).toEqual(200);
      expect(json.frame.height).toEqual(100);
      expect(json.frame.x).toEqual(0);
      expect(json.frame.y).toEqual(0);
    });
    it('不带参数生成', () => {
      const page = new Page();
      const json = page.toSketchJSON();
      expect(json._class).toBe('page');
      expect(json.frame.width).toEqual(0);
      expect(json.frame.height).toEqual(0);
      expect(json.frame.x).toEqual(0);
      expect(json.frame.y).toEqual(0);
    });
  });
});
