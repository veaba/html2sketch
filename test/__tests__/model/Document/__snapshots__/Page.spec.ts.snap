// Vitest Snapshot v1

exports[`Page 类 > Page 创建 > 不带参数创建 1`] = `
{
  "_class": "page",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "BA0CE274-426F-4E90-8F57-2A5FF3A81C0F",
  "exportOptions": {
    "_class": "exportOptions",
    "exportFormats": [],
    "includedLayerIds": [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClickThrough": true,
  "hasClippingMask": false,
  "horizontalRulerData": {
    "_class": "rulerData",
    "base": 0,
    "guides": [],
  },
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isTemplate": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": [],
  "name": "page",
  "nameIsFixed": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": {
    "_class": "style",
    "borderOptions": {
      "_class": "borderOptions",
      "dashPattern": [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": [],
    "colorControls": {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "15242FC6-2D35-4273-BDB4-3B6B56DA9282",
    "endMarkerType": 0,
    "fills": [],
    "innerShadows": [],
    "miterLimit": 10,
    "shadows": [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "userInfo": null,
  "verticalRulerData": {
    "_class": "rulerData",
    "base": 0,
    "guides": [],
  },
}
`;

exports[`Page 类 > Page 创建 > 带参数创建 1`] = `
{
  "_class": "page",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "468E2AEA-8CBB-48A9-A9D8-89A153EC8F57",
  "exportOptions": {
    "_class": "exportOptions",
    "exportFormats": [],
    "includedLayerIds": [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": {
    "_class": "rect",
    "constrainProportions": false,
    "height": 100,
    "width": 200,
    "x": 0,
    "y": 0,
  },
  "hasClickThrough": true,
  "hasClippingMask": false,
  "horizontalRulerData": {
    "_class": "rulerData",
    "base": 0,
    "guides": [],
  },
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isTemplate": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": [],
  "name": "page",
  "nameIsFixed": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": {
    "_class": "style",
    "borderOptions": {
      "_class": "borderOptions",
      "dashPattern": [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": [],
    "colorControls": {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "CDE5FB61-3E19-4511-8D0C-3203EA02D20F",
    "endMarkerType": 0,
    "fills": [],
    "innerShadows": [],
    "miterLimit": 10,
    "shadows": [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "userInfo": null,
  "verticalRulerData": {
    "_class": "rulerData",
    "base": 0,
    "guides": [],
  },
}
`;

exports[`Page 类 Page 创建 不带参数创建 1`] = `
Object {
  "_class": "page",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "UUID",
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClickThrough": true,
  "hasClippingMask": false,
  "horizontalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
  "includeInCloudUpload": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [],
  "name": "page",
  "nameIsFixed": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "UUID",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "userInfo": null,
  "verticalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
}
`;

exports[`Page 类 Page 创建 带参数创建 1`] = `
Object {
  "_class": "page",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "UUID",
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 100,
    "width": 200,
    "x": 0,
    "y": 0,
  },
  "hasClickThrough": true,
  "hasClippingMask": false,
  "horizontalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
  "includeInCloudUpload": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [],
  "name": "page",
  "nameIsFixed": false,
  "resizingConstraint": 63,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": Object {
    "_class": "style",
    "borderOptions": Object {
      "_class": "borderOptions",
      "dashPattern": Array [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": Array [],
    "colorControls": Object {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": Object {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "UUID",
    "endMarkerType": 0,
    "fills": Array [],
    "innerShadows": Array [],
    "miterLimit": 10,
    "shadows": Array [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "userInfo": null,
  "verticalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
}
`;
